### PR TITLE
fix: cleanup ini parser logic with better fallbacks

### PR DESCRIPTION
**Pull request by elibosley**
Added a new parser for INI boolean values, including functions to convert INI boolean strings to JavaScript booleans and handle malformed inputs. Introduced unit tests to validate the functionality of both `iniBooleanToJsBoolean` and `iniBooleanOrAutoToJsBoolean`, covering various valid, malformed, and edge case scenarios. Updated state parsers to utilize the new functions for improved reliability.

### CODE REVIEW

File: api/src/core/utils/parsers/ini-boolean-parser.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        }
+        if (cleanValue === 'yes' || cleanValue === 'true') {
+            return true;
+        }
+    }
+
+    // Always return defaultValue when provided (even if undefined)
+    if (arguments.length >= 2) {
+        return defaultValue;
+    }
```
Comment (pujitm) [inline api/src/core/utils/parsers/ini-boolean-parser.ts:43]: i'm prob missing something -- is `arguments` declared?

Comment (elibosley) [reply api/src/core/utils/parsers/ini-boolean-parser.ts:43]: ⏺ In this context, arguments is a special JavaScript object that contains all the arguments passed to the function. Here's
  what it would be set to in different scenarios:

  1. When called with one argument: iniBooleanToJsBoolean("yes")
    - arguments.length = 1
    - arguments[0] = "yes"
  2. When called with two arguments: iniBooleanToJsBoolean("invalid", false)
    - arguments.length = 2
    - arguments[0] = "invalid"
    - arguments[1] = false
  3. When called with explicit undefined: iniBooleanToJsBoolean("invalid", undefined)
    - arguments.length = 2
    - arguments[0] = "invalid"
    - arguments[1] = undefined

  The check arguments.length >= 2 determines if a second parameter was actually passed, regardless of its value. This allows
  the function to distinguish between:
  - No default provided (return undefined)
  - Default explicitly set to undefined (return undefined as the default)
